core-bundle/src/InsertTag/Resolver/DateInsertTag.php

```diff
@@ -21,14 +21,31 @@
 #[AsInsertTag('date', asFragment: true)]
 class DateInsertTag implements InsertTagResolverNestedResolvedInterface
 {
+    private const CACHEABLE_FORMAT_CHARACTERS = ['Y', 'm', 'd'];
+
     public function __invoke(ResolvedInsertTag $insertTag): InsertTagResult
     {
         $format = $insertTag->getParameters()->get(0) ?? $GLOBALS['objPage']->dateFormat ?? $GLOBALS['TL_CONFIG']['dateFormat'] ?? '';
         $result = new InsertTagResult(Date::parse($format), OutputType::text);
 
-        // Special handling for the very common {{date::Y}} (e.g. in the website footer) case
-        if ('Y' === $format) {
-            $result = $result->withExpiresAt(new \DateTimeImmutable($result->getValue().'-12-31 23:59:59'));
+        preg_match_all('/['.implode('', self::CACHEABLE_FORMAT_CHARACTERS).']/', $format, $matches);
+        $usedFormatChars = $matches[0] ?? [];
+        sort($usedFormatChars);
+
+        // Add caching headers for supported formats
+        if (0 === \count(array_diff($usedFormatChars, self::CACHEABLE_FORMAT_CHARACTERS))) {
```

Comment:
Yeah this happened because I've changed concept like 4 times. I need to know all characters that are supported by `Date` to only extract those and then diff agains what we support. :)

Reply:
Fixed.